packages/core/src/timeout/__tests__/timeout.test.ts

```diff
@@ -0,0 +1,117 @@
+import { describe, test, vi, expect } from 'vitest';
+import {
+  allSettled,
+  createEvent,
+  createStore,
+  createWatch,
+  fork,
+} from 'effector';
+
+import { createQuery } from '../../query/create_query';
+import { timeout } from '../timeout';
+import { isTimeoutError } from '../../errors/guards';
+
+describe('timeout(query, time)', () => {
```

Comment:
🚧 Could you use fake timers for these tests, please? In the case of event loop lags, they can fail because of longer delays between tasks.